### PR TITLE
Update module configuration

### PR DESCRIPTION
**Pull request by gantunesr**
## Description

This PR updates the general configuration of the KeyringController to module to match the ones from https://github.com/MetaMask/metamask-module-template.

### CODE REVIEW

**Comment by gantunesr** [comment]
@SocketSecurity ignore-all

File: .github/workflows/build-lint-test.yml [hunk 1]
```diff
@@ -80,7 +80,7 @@ jobs:
       - prepare
     strategy:
       matrix:
-        node-version: [14.x, 16.x, 18.x, 19.x]
+        node-version: [16.x, 18.x, 20.x]
```
[the remark this answers is not in the document]
Comment (gantunesr) [reply .github/workflows/build-lint-test.yml]: There's an issue when trying to run the workflow in Node 14

```
Run actions/setup-node@v3
Attempting to download 14x...
Not found in manifest. Falling back to download directly from Node
Error: Unable to find Node version '14x' for platform linux and architecture x64.
```

[the remark this answers is not in the document]
Comment (gantunesr) [reply .github/workflows/build-lint-test.yml]: Ah just saw your other comment

File: .github/workflows/build-lint-test.yml [hunk 2]
```diff
@@ -24,7 +24,7 @@ jobs:
...
```
[the remark this answers is not in the document]
Comment (gantunesr) [reply .github/workflows/build-lint-test.yml]: @legobeat do you know how to solve this issue? I tried to add a `clear cache` step but it didn't work and I don't see what part of the configuration in the following files `package.json`, `.nvmrc`, `main.yml`, and `build-lint-test.yml`

[the remark this answers is not in the document]
Comment (gantunesr) [reply .github/workflows/build-lint-test.yml]: Minor typo

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -47,3 +48,30 @@ jobs:
...
+  publish-release:
+    needs: is-release
+    if: needs.is-release.outputs.IS_RELEASE == 'true'
+    name: Publish release
+    permissions:
+      contents: write
+    uses: ./.github/workflows/publish-release.yml
+    secrets:
+      NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+      PUBLISH_DOCS_TOKEN: ${{ secrets.PUBLISH_DOCS_TOKEN }}
```
Comment (Gudahtt) [inline .github/workflows/main.yml:76]: Missing this repo secret

Comment (Gudahtt) [inline .github/workflows/main.yml:76]: Missing this repo secret

Comment (Gudahtt) [reply .github/workflows/main.yml:76]: The secret has been added, doc publishing should be working now

**Pull request merged**